### ISSUE TITLE
Translation of XmlJavaTypeAdapter doesn't work like expected

### ISSUE THREAD

**Issue by CSchulz** [FasterXML/jackson-module-jaxb-annotations#43]
Hello,

I have recognized with webron together (swagger team) that there is an issue with the translation of XmlJavaTypeAdapter to json.

I had registered the needed module (JaxbAnnotationModule) and generated the json.

The swagger json looks like

```
    "PackageNumber" : {
      "type" : "object",
      "properties" : {
        "value" : {
          "type" : "string"
        }
      }
    },
```

The same in xsd looks like

```
<xs:element minOccurs="0" name="packageNumber" type="xs:string"/>
```

You can  see the xsd element is correct generated.

Some parts from the relevant classes:

```
@XmlAccessorType(XmlAccessType.FIELD)
@XmlRootElement(namespace = "http://namespace/v1")
@AssociationOverride(...)
public class FloatPackage extends AbstractItem {

    @NotNull
    @AttributeOverride(name = "value", column = @Column(name = "FGP_NUMBER", unique = true))
    private PackageNumber packageNumber;
```

The class PackageNumber:

```
@Embeddable
@XmlJavaTypeAdapter(PackageNumber.Adapter.class)
public class PackageNumber extends AbstractValueObject<String> {
```

AbstractValueObject:

```
@MappedSuperclass
public abstract class AbstractValueObject<V> {
    @Expose
    private V value;
```

Whole discussion with webron:
https://github.com/swagger-api/swagger-core/issues/960

**Comment by cowtowncoder** [comment]
I'll read through the discussion, but could you quickly summarize here the actual problem?
I am aware that there are issues in handling of adapters in general, but not sure which aspects is incorrect here (serialization/deserialization of JSON, Schema generation, POJO introspection?)

**Comment by cowtowncoder** [comment]
Ok, so specifically, a simple self-contained example would be very useful. I have no doubt that this module is missing (or mis-implementing) support, just want to know specific parts to hopefully be able to fix the issue.

**Comment by CSchulz** [comment]
It seems like the adapter implementation is "ignored" by Jackson. It means the **value** is handled as property of the adapter instead of the *PackageNumber* class.

Compare the json output with the xsd output. You can see that the packageNumber is directly a string and doesn't have a child element called **value**.

**Comment by cowtowncoder** [comment]
With a simple reproducible test case it should be easy to see what happens, if anyone has time to work on this.

**Comment by CSchulz** [comment]
How do you prefer the test case? What would be the best practice?

**Comment by CSchulz** [comment]
Here is an example:

https://github.com/CSchulz/xml-adapter-showcase

swagger accessible /test/rest/swagger.json.

**Comment by cowtowncoder** [comment]
Thank you! As to how to contribute test cases; inclusion within issue is fine for smaller cases, separate project also works, as does a pull request. We are not picky about details.
I hope to have a look at the project, but since I just returned after being offline for a week there's bit of backlog, so there may be more delay than usual.

**Comment by cowtowncoder** [comment]
@CSchulz Going back to basics, one quick question: is the problem with simple JSON serialization of values; or for JSON Schema generation of class hierarchy? I have so far assumed former, but since the output looks like JSON Schema definition, it could also be about generation of JSON Schema (and serialization from that).

Also: I understand that reproduction is possible with the service, but unfortunately that does not quite explain what the problem is to me. I can just see the output, but not quite understand why it should be something else.

**Comment by mdzhigarov** [comment]
Hello,

I've created a very simple self-contained example:
https://github.com/mdzhigarov/swagger-jackson-test

Please notice that at the end, the swagger.json contains:
"Link" : {
      "type" : "object",
      "properties" : {
        "type" : {
          "type" : "string"
        },
        "params" : {
          "type" : "object",
          "additionalProperties" : {
            "type" : "string"
          }
        },
        "uri" : {
          "type" : "string",
          "format" : "uri"
        },
        "rels" : {
          "type" : "array",
          "items" : {
            "type" : "string"
          }
        },
        "uriBuilder" : {
          "$ref" : "#/definitions/UriBuilder"
        },
        "rel" : {
          "type" : "string"
        },
        "title" : {
          "type" : "string"
        }
      }

But I'd expect the model to look like javax.ws.rs.core.Link$JaxbLink instead - having only a "href" attribute & probably some rels.

This issue is also related to: https://github.com/kongchen/swagger-maven-plugin/issues/246

**Comment by cowtowncoder** [comment]
@mdzhigarov thank you.

**Comment by cowtowncoder** [comment]
@mdzhigarov Hmmh. That is certainly quite a minimal piece of code, and I am not quite sure how to invoke that to see the issue.

But one declaration that I see which is probably problematic is:

```java
    @XmlElement(name = "link")
    @XmlJavaTypeAdapter(Link.JaxbAdapter.class)
    List<Link> links = new ArrayList<Link>();
```

This is mostly since Jackson has no knowledge of JAX-RS types, either `JaxbLink` or `Link` itself.
So what is probably needed is custom `JsonSerializer`s for `Link` and getting rid of Adapter altogether as it won't help a lot here as resulting type is still not something Jackson recognizes, except as general POJO type.

**Comment by cowtowncoder** [comment]
Ok, maybe this:

https://github.com/FasterXML/jackson-jaxrs-providers/issues/48

is exactly what we need. That is for `Link` deserialization. Of course, use of JAX-RS provider is different from having, say, `jackson-datatype-jaxrs`. But maybe I can start by actually creating datatype lib first, under JAX-RS providers, and then that can be reusable outside provider.

**Comment by CSchulz** [comment]
@cowtowncoder haven't seen your question. It is more about the schema generation which differs (see my initial post).

**Comment by cowtowncoder** [comment]
@CSchulz yes, but I was trying to figure out later submitted test what was failing.

I guess I still don't quite have reproduction for the original issue.

**Comment by CSchulz** [comment]
What do you need to get more details?

**Comment by cowtowncoder** [comment]
@CSchulz a self-contained unit test would be best.

**Comment by cowtowncoder** [comment]
Fwtw, https://github.com/FasterXML/jackson-jaxrs-providers/issues/48 is now implemented for 2.8.0, so registering `jackson-datatype-jaxrs` will add support for reading/writing `Link` values regardless of JAXB annotations. It will consider type to be `String`, however, since that is how it will be serialized; and not as Bean with properties.

**Comment by cowtowncoder** [comment]
At this point project has moved, so if the issue exists, would need to be re-created with up-to-date explanation at:

https://github.com/FasterXML/jackson-modules-base/issues
